### PR TITLE
fix(s3-notifications): notifications allowed with imported kms keys

### PR DESCRIPTION
**Pull request by markussiebert**
----
fixes: #18988

If you add an sqs queue as notification target to an s3 bucket, and this sqs queue is encrypted with an imported kms IKey, the stack won't synthesize. Instead of failing, it should warn the user, that it can not ensure the correct kms key policy permissions.

This fix will solve this.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-s3-notifications/lib/sqs.ts
Comment (peterwoodworth) [inline packages/@aws-cdk/aws-s3-notifications/lib/sqs.ts:40]: Typo here "Ensure hat" 🙂

**Review by peterwoodworth — CHANGES_REQUESTED** [review]
Looks great to me @markussiebert, just a small typo!

File: packages/@aws-cdk/aws-s3-notifications/lib/sqs.ts
Comment (markussiebert) [reply packages/@aws-cdk/aws-s3-notifications/lib/sqs.ts]: Thank you so much! Fixed this

**Comment by peterwoodworth** [comment]
Hey @markussiebert, you don't have to worry about merging master in. That will happen automatically once we approve it 😄

**Review by peterwoodworth — APPROVED** [review]
Looks good to me! @otaviomacedo is the owner here so I'll let him take a final look before merging 🙂

**Review by peterwoodworth — DISMISSED** [review]
Love the optimization you've made here. Clever! Still looks great to me

**Review by otaviomacedo — APPROVED** [review]

**Review by otaviomacedo — APPROVED** [review]

**Pull request merged**
